crates/net/discv5/src/filter.rs

```diff
@@ -0,0 +1,96 @@
+//! Predicates to constraint peer lookups.
+
+use derive_more::Constructor;
+
+/// Allows users to inject custom filtering rules on which peers to discover.
+pub trait FilterDiscovered {
+    /// Applies filtering rules on [`Enr`](discv5::Enr) data. Returns [`Ok`](FilterOutcome::Ok) if
+    /// peer should be included, otherwise [`Ignore`](FilterOutcome::Ignore).
+    fn filter_discovered_peer(&self, _enr: &discv5::Enr) -> FilterOutcome {
+        FilterOutcome::Ok
+    }
+
+    /// Message for [`FilterOutcome::Ignore`] should specify the reason for filtering out a node
+    /// record.
+    fn ignore_reason(&self) -> String;
```

Comment:
hmm, I don't think this makes much sense as part of the trait, because this is only used to populate the return value of `filter_discovered_peer` which is already part of the trait

Reply:
it is reused in the trait below, and encourages implementations of this trait to be as simple as possible, i.e. for the call to `filter_discovered_peers` to only return on one condition: the one producing the ignore reason. can move it out of the trait though if it bothers you?